### PR TITLE
XMPP recipe

### PR DESCRIPTION
**Pull request by Aman9das**
This basically compiles 2 different projects.. kindly review the PR

### CODE REVIEW

File: recipes/xmpp/webview.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+// xmpp integration
+/*jshint esversion: 6 */
+module.exports = (Franz) => {
+    function getMessages() {
+        let direct = 0;
+        let indirect = 0;
+        document.querySelectorAll('.msgs-indicator').forEach(indicator => {
+            direct += parseInt(indicator.innerText, 10)
```
Comment (vraravam) [inline recipes/xmpp/webview.js:8]: Please use `Ferdi.safeParseInt` for parsing text into numbers.

File: recipes/xmpp/webview.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+        })
+        
+        direct = direct / 2 // as the messages are provided in 2 different locations..
+        Franz.setBadge(direct, indirect);
```
Comment (vraravam) [inline recipes/xmpp/webview.js:12]: If `indirect` is always zero, you can remove the references.

**Review by vraravam — CHANGES_REQUESTED** [review]
Please remove the `png` file from the PR.

**Comment by Aman9das** [comment]
Thanks a lot! Hopefully this is ok now :)

File: recipes/xmpp/package.json [hunk 3]
```diff
@@ -0,0 +1,14 @@
+{
+    "id": "xmpp",
```
Comment (vraravam) [inline recipes/xmpp/package.json:2]: Please adhere to the formatting of 2 spaces, no tabs

File: recipes/xmpp/webview.js [hunk 4]
```diff
@@ -0,0 +1,16 @@
+// xmpp integration
+/*jshint esversion: 6 */
+module.exports = (Ferdi) => {
+    function getMessages() {
+        let direct = 0;
+        let indirect = 0;
```
Comment (vraravam) [inline recipes/xmpp/webview.js:6]: Please remove the unused variable.

**Review by vraravam — CHANGES_REQUESTED** [review]

File: recipes/xmpp/webview.js [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+        document.querySelectorAll('.msgs-indicator').forEach(indicator => {
+            direct += Ferdi.safeParseInt(indicator.innerText, 10)
```
Comment (vraravam) [inline recipes/xmpp/webview.js:8]: There's no 2nd argument for `Ferdi.safeParseInt`

**Review by vraravam — CHANGES_REQUESTED** [review]

**Comment by vraravam** [comment]
@Aman9das - please follow the instructions in the README while contributing a new recipe. There's a step related to running the `pnpm` command. This would have modified the `all.json` file - which needs to be included in the PR. I don't see that change. Please follow the steps and add all the changed files.

File: recipes/toggl/webview-unsafe.js [hunk 6]
```diff
@@ -1,11 +0,0 @@
-// only try to update badge once Ferdi API has finished loading
```
Comment (vraravam) [inline recipes/toggl/webview-unsafe.js:1]: Why are you deleting code for a recipe that isn't related to this PR?

File: recipes/xmpp/webview.js [hunk 7]
```diff
@@ -3,10 +3,10 @@
 module.exports = (Ferdi) => {
     function getMessages() {
         let direct = 0;
-        let indirect = 0;
-        document.querySelectorAll('.msgs-indicator').forEach(indicator => {
-            direct += Ferdi.safeParseInt(indicator.innerText, 10)
-        })
+        // let indirect = 0; //will be used in a future verios
```
Comment (vraravam) [inline recipes/xmpp/webview.js:6]: We dont need this for now - please remove. Its just noise.

**Review by vraravam — CHANGES_REQUESTED** [review]

**Comment by vraravam** [comment]
Please rebase from the `master` branch of the upstream - so that the issues with the `toggl` recipe are fixed.

File: recipes/toggl/webview-unsafe.js (hunk 6 above)
Comment (Aman9das) [reply recipes/toggl/webview-unsafe.js:1]: OH i forgot to tell, theres an issue with that recipe that is preventing git commit..

File: recipes/xmpp/webview.js (hunk 7 above)
Comment (Aman9das) [reply recipes/xmpp/webview.js:6]: Documentation, it won't afffect performance, please just let it be. Documentation in any form is helpful for the next person down the line

**Comment by Aman9das** [comment]
Please create some other branch to rebase to.. I will rebase.

**Comment by vraravam** [comment]
> Please create some other branch to rebase to.. I will rebase.

Your PR should have been against a different branch in your fork. Please see the published docs from Github.

The CI process has not thrown any error with the `toggl` recipe. Please disregard anything that might show up on your local machine.

File: recipes/xmpp/webview.js (hunk 7 above)
Comment (vraravam) [reply recipes/xmpp/webview.js:6]: Though I agree in general, but, this line (imo) is not "documentation". I can accept this for now.

File: recipes/toggl/webview-unsafe.js (hunk 6 above)
Comment (vraravam) [reply recipes/toggl/webview-unsafe.js:1]: Please undo this change - cannot be compromised for this PR.

**Review by vraravam — CHANGES_REQUESTED** [review]
The changes to the `toggl` recipe should be undone.

**Comment by Aman9das** [comment]
The toggl webview-unsafe 'ferdi undefined' error is not letting me commit. the data is present in the pr, kindly try to implement the recipe properly on own device if possible.

Closing the pr as can't satisfy the scripts

**Comment by vraravam** [comment]
Closing this PR - since the code has been merged asynchronously (with attribution to @Aman9das )

**Pull request closed without merging**